File: lib/model/block.dart
import 'package:pos_printer/model/enum.dart';
import 'package:json_annotation/json_annotation.dart';

part 'block.g.dart';

@JsonSerializable()
class Block {
  Block({
    this.content = '',
    this.type = POSType.text,
    this.align = POSAlign.left,
    this.font = POSFont.medium,
    this.fontWeight = POSFontWeight.normal,
    this.underline = POSUnderline.none,
    this.columnWidth = POSWidth.full,
  });

  String content;
  POSType type;
  POSAlign align;
  POSFont font;
  POSFontWeight fontWeight;
  POSUnderline underline;
  POSWidth columnWidth;

  factory Block.fromJson(Map<String, dynamic> json) => _$BlockFromJson(json);
  Map<String, dynamic> toJson() => _$BlockToJson(this);
}

File: android/src/main/kotlin/com/tokoin/pos_printer/model/Block.kt
package com.tokoin.pos_printer.model

import org.json.JSONObject

class Block(json: String): JSONObject(json) {
    val content: String = this.optString("content") ?: ""
    val align: POSAlign = enumValueOrNull(this.optString("align")) ?: POSAlign.Left
    val font: POSFont = enumValueOrNull(this.optString("font")) ?: POSFont.Medium
    val fontWeight: POSFontWeight = enumValueOrNull(this.optString("weight")) ?: POSFontWeight.Normal
    val underline: POSUnderline = enumValueOrNull(this.optString("underline")) ?: POSUnderline.None
    val columnWidth: POSWidth = enumValueOrNull(this.optString("width")) ?: POSWidth.Full
}